eng/scripts/generate_from_source_pom.py

```diff
@@ -101,14 +102,14 @@ def create_from_source_pom(project_list: str, set_pipeline_variable: str):
     for project_identifier in dependent_modules:
         dependency_modules = resolve_project_dependencies(project_identifier, dependency_modules, projects)
 
-    modules: Set[str] = set()
+    source_projects: Set[Project] = set()
 
-    # Finally map the project identifiers to relative module paths.
-    add_module_paths(modules, project_list_identifiers, projects)
-    add_module_paths(modules, dependent_modules, projects)
-    add_module_paths(modules, dependency_modules, projects)
-
-    modules = sorted(modules)
+    # Finally map the project identifiers to projects.
+    add_source_projects(source_projects, project_list_identifiers, projects)
+    add_source_projects(source_projects, dependent_modules, projects)
+    add_source_projects(source_projects, dependency_modules, projects)
+    
+    modules = list(set(sorted([p.module_path for p in source_projects])))
```

Comment:
I think you sort/set/list this variable three times. Might be simpler to do it once?